pkgs/development/python-modules/patool/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv, lib, buildPythonApplication, fetchFromGitHub, pytest, p7zip, unzip,
+  cabextract, zip, zopfli, lzip, zpaq, gnutar, gnugrep, diffutils, file,
+  gzip, bzip2, xz}:
+
+# unrar is unfree, as well as 7z with unrar support, not including it (patool doesn't support unar)
+
+let
+  compression-utility = [
+    p7zip
+    unzip
+    gnutar
+    cabextract
+    zip
+    zopfli
+    lzip
+    zpaq
+    gzip
+    gnugrep
+    diffutils
+    bzip2
+    file
+    xz
+  ];
+in
+buildPythonApplication rec {
+  pname = "patool";
+  version = "1.12";
+
+  #pypi doesn't have test data
+  src = fetchFromGitHub {
+    owner = "wummel";
+    repo = pname;
+    rev = "upstream/${version}";
+    sha256 = "Xv4aCUnLi+b1T29tuKRADTIWwK2dO8iDP/D7UfU5mWw=";
```

Comment:
There's still a preference to having base32 hashes within nixpkgs to ease the upgrade process for users with holder versions of the nix cli.

You can use `nix-hash --to-base32 <hash>` to convert the hash into the related base32 hash.

For more details refer to: https://github.com/NixOS/nixpkgs/pull/89308#issuecomment-636927054 #89275